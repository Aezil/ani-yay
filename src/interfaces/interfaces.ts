export enum MediaType {
  anime = 'ANIME',
  manga = 'MANGA',
}

export enum MediaStatus {
  finished = 'FINISHED',
  releasing = 'RELEASING',
  notYetReleased = 'NOT_YET_RELEASED',
  cancelled = 'CANCELLED',
  hiatus = 'HIATUS',
}

export interface Media {
  id: number;
  title: {
    romaji: string;
  };
  description: string;
  status: MediaStatus;
  type: MediaType;
  startDate: {
    year: number;
    month: number;
    day: number;
  };
  endDate: {
    year: number;
    month: number;
    day: number;
  };
  popularity: number;
  format: string;
}

export interface PageInfo {
  total: number;
  currentPage: number;
  lastPage: number;
  hasNextPage: boolean;
  perPage: number;
}

export interface AnimeQueryResult {
  Page: {
    pageInfo: PageInfo;
    media: Media[];
  };
}
