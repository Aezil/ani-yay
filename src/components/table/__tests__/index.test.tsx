import { render } from '@testing-library/react';
import React from 'react';

import { Media, MediaStatus, MediaType } from '../../../interfaces/interfaces';
import { AniYayTable } from '..';

const MEDIA_MOCK: Media[] = [
  {
    id: 3423,
    title: {
      romaji: 'Woo',
    },
    description: 'I am a description',
    status: MediaStatus.cancelled,
    type: MediaType.anime,
    popularity: 5,
    startDate: {
      day: 1,
      month: 1,
      year: 2022,
    },
    endDate: {
      day: 1,
      month: 1,
      year: 2022,
    },
    format: 'TV',
  },
  {
    id: 4,
    title: {
      romaji: 'Nice',
    },
    description: 'I am a description',
    status: MediaStatus.cancelled,
    type: MediaType.anime,
    popularity: 5,
    startDate: {
      day: 1,
      month: 1,
      year: 2022,
    },
    endDate: {
      day: 1,
      month: 1,
      year: 2022,
    },
    format: 'TV',
  },
];

describe('Ani-Yay Table', () => {
  it('should render table component with no value', () => {
    const { container } = render(<AniYayTable media={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the table with a few values', () => {
    const { container } = render(<AniYayTable media={MEDIA_MOCK} />);
    expect(container).toMatchSnapshot();
  });
});
