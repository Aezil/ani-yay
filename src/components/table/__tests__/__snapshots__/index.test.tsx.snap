// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ani-Yay Table should render table component with no value 1`] = `
<div>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Type
        </th>
        <th>
          Popularity
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Ani-Yay Table should render the table with a few values 1`] = `
<div>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Type
        </th>
        <th>
          Popularity
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Woo
        </td>
        <td>
          I am a description
        </td>
        <td>
          CANCELLED
        </td>
        <td>
          ANIME
        </td>
        <td>
          5
        </td>
      </tr>
      <tr>
        <td>
          Nice
        </td>
        <td>
          I am a description
        </td>
        <td>
          CANCELLED
        </td>
        <td>
          ANIME
        </td>
        <td>
          5
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
